packages/rpc/src/rpcHandler.ts

```diff
@@ -910,4 +816,183 @@ export class RpcHandler implements IRpcEndpoint {
+                    `maxPriorityFeePerGas must be at least ${minMaxFeePerGas} (current maxPriorityFeePerGas: ${gasPrice.maxPriorityFeePerGas}) - use pimlico_getUserOperationGasPrice to get the current gas price`
+                )
+            }
+        }
+
+        if (userOperation.verificationGasLimit < 10000n) {
+            throw new RpcError("verificationGasLimit must be at least 10000")
+        }
+
+        this.logger.trace({ userOperation, entryPoint }, "beginning validation")
+        this.metrics.userOperationsReceived.inc()
+
+        if (
+            userOperation.preVerificationGas === 0n ||
+            userOperation.verificationGasLimit === 0n ||
+            userOperation.callGasLimit === 0n
+        ) {
+            throw new RpcError(
+                "user operation gas limits must be larger than 0"
+            )
+        }
+
+        const entryPointContract = getContract({
+            address: this.entryPoint,
+            abi: EntryPointAbi,
+            publicClient: this.publicClient
+        })
+
+        const [nonceKey, userOperationNonceValue] = getNonceKeyAndValue(
+            userOperation.nonce
+        )
+
+        const getNonceResult = await entryPointContract.read.getNonce(
+            [userOperation.sender, nonceKey],
+            {
+                blockTag: "latest"
+            }
+        )
+
+        const [_, currentNonceValue] = getNonceKeyAndValue(getNonceResult)
+
+        if (userOperationNonceValue < currentNonceValue) {
+            throw new RpcError(
+                "UserOperation reverted during simulation with reason: AA25 invalid account nonce",
+                ValidationErrors.SimulateValidation
+            )
+        }
+        if (userOperationNonceValue > currentNonceValue + 10n) {
+            throw new RpcError(
+                "UserOperation reverted during simulation with reason: AA25 invalid account nonce",
+                ValidationErrors.SimulateValidation
+            )
+        }
+        if (userOperationNonceValue === currentNonceValue) {
+            const validationResult =
+                await this.validator.validateUserOperation(userOperation)
+            await this.reputationManager.checkReputation(
+                userOperation,
+                validationResult
+            )
+            await this.mempool.checkEntityMultipleRoleViolation(userOperation)
+            const success = this.mempool.add(
+                op,
+                validationResult.referencedContracts
+            )
+            if (!success) {
+                throw new RpcError(
+                    "UserOperation reverted during simulation with reason: AA25 invalid account nonce",
+                    ValidationErrors.SimulateValidation
+                )
+            }
+        } else {
+            this.nonceQueuer.add(userOperation)
+        }
+    }
+
+    async pimlico_sendCompressedUserOperation(
+        compressedCalldata: Hex,
+        entryPoint: Address,
```

Comment:
hmm could we make the inflatorAddress second, and the entryPoint last here?

since sendUserOperation also has the entryPoint last